pyseir/deployment/demographic_mapper.py

```diff
@@ -0,0 +1,729 @@
+        compartments.
+
+        Returns
+        -------
+        hospitalization_inflow_rates: dict
+            Contains:
+            - <hospitalization>: : np.array
+              Age specific per-capita rate of admission to corresponding
+              hospitalization.
+            hospitalization include: HGen, HICU, HVent.
+        """
+        hospitalization_inflow_rates = {}
+        hospitalization_inflow_rates['HGen'] = \
+            (self.parameters['hospitalization_rate_general']
+           - self.parameters['hospitalization_rate_icu']) / self.parameters['symptoms_to_hospital_days']
+        hospitalization_inflow_rates['HICU'] = self.parameters['hospitalization_rate_icu'] / self.parameters[
+            'symptoms_to_hospital_days']
+        hospitalization_inflow_rates['HVent'] = \
+            hospitalization_inflow_rates['HICU'] * self.parameters['fraction_icu_requiring_ventilator']
+
+        return hospitalization_inflow_rates
+
+    def _reconstruct_hospital_recovery_inflow_rates(self):
+        """
+        Reconstruct trajectory of inferred per capita rates of recovery
+        during hospitalization through time from MLE model parameters and
+        compartments.
+
+        Returns
+        -------
+        hospital_recovery_inflow_rates: dict
+            Contains:
+            - <hospitalization>: np.array, age-specific rate of recovery
+                                 from the corresponding hospitalization.
+              hospitalization include: HGen, HICU, HVent.
+        """
+        mortality_inflow_rates = self._reconstruct_mortality_inflow_rates()
+
+        hospital_recovery_inflow_rates = {}
+        hospital_recovery_inflow_rates['HGen'] = (1 - mortality_inflow_rates['HGen']) / \
+            self.parameters['hospitalization_length_of_stay_general']
+        hospital_recovery_inflow_rates['HICU'] = (1 - mortality_inflow_rates['HICU']) * (1 - self.parameters[
+            'fraction_icu_requiring_ventilator']) \
+            / self.parameters['hospitalization_length_of_stay_icu']
+        hospital_recovery_inflow_rates['HVent'] = \
+            (1 - np.maximum(mortality_inflow_rates['HVent'],
+                            self.parameters['mortality_rate_from_ICUVent'])) \
+            / self.parameters['hospitalization_length_of_stay_icu_and_ventilator']
+
+        return hospital_recovery_inflow_rates
+
+    def _age_specific_IHR_by_hospitalization(self, measure_unit):
+        """
+        Calculate age specific hospitalization rates among infected
+        population (IHR) for a given measure unit by types of
+        hospitalization.
+
+        When unit is per capita, calculates the probability that an infected
+        person (asymptomatic or symptomatic) ultimately gets admitted to
+        hospital:
+            probability of being symptomatic * rate of hospitalized / (rate of
+            hospitalized + rate of recovery without hospitalization)
```

Comment:
Apologies if I'm missing something, but I don't see this being computed in the code

Reply:
Ah the conditions should be switched here. Thanks for pointing this out!